corehq/apps/dump_reload/sql/load.py

```diff
@@ -153,35 +200,27 @@ def load_data_for_db(db_alias, objects):
     return LoadStat(db_alias, model_counter)
 
 
-def _group_objects_by_db(objects):
-    """
-    :param objects: Deserialized object dictionaries
-    :return: List of tuples of (db_alias, [object,...])
-    """
-    objects_by_db = defaultdict(list)
-    for obj in objects:
-        app_label = obj['model']
-        model = apps.get_model(app_label)
-        router_hints = {}
-        if hasattr(model, 'partition_attr'):
+def get_db_alias(obj: dict) -> str:
+    app_label = obj['model']
+    model = apps.get_model(app_label)
+    router_hints = {}
+    if hasattr(model, 'partition_attr'):
+        try:
+            partition_value = obj['fields'][model.partition_attr]
+        except KeyError:
+            # in the case of foreign keys the serialized field name is the
+            # name of the foreign key attribute
+            field = [
+                field for field in model._meta.fields
+                if field.column == model.partition_attr
+            ][0]
```

Comment:
Slightly more defensive. I'm assuming there should always be exactly one.
```suggestion
            field, = [
                field for field in model._meta.fields
                if field.column == model.partition_attr
            ]
```